src/pages/registration/contributors_tab/Authors.tsx

```diff
@@ -0,0 +1,130 @@
+import deepmerge from 'deepmerge';
+import { FieldArrayRenderProps, FormikProps, useFormikContext, move } from 'formik';
+import React, { FC, useState, useEffect } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useDispatch } from 'react-redux';
+import { setNotification } from '../../../redux/actions/notificationActions';
+import { Authority } from '../../../types/authority.types';
+import { Contributor, emptyContributor, UnverifiedContributor } from '../../../types/contributor.types';
+import { NotificationVariant } from '../../../types/notification.types';
+import { BackendTypeNames } from '../../../types/publication_types/commonRegistration.types';
+import { Registration } from '../../../types/registration.types';
+import { overwriteArrayMerge } from '../../../utils/formik-helpers';
+import AuthorTable from './components/AuthorTable';
+import AddIcon from '@material-ui/icons/Add';
+import styled from 'styled-components';
+import { Button } from '@material-ui/core';
+import AddContributorModal from './AddContributorModal';
+
+const StyledAddAuthorButton = styled(Button)`
+  margin-top: 1rem;
+`;
+
+interface AuthorsProps extends Pick<FieldArrayRenderProps, 'push' | 'replace'> {}
+
+const Authors: FC<AuthorsProps> = ({ push, replace }) => {
+  const { t } = useTranslation('registration');
+  const dispatch = useDispatch();
+  const { values, setValues }: FormikProps<Registration> = useFormikContext();
+  const {
+    entityDescription: { contributors },
+  } = values;
+  const [openContributorModal, setOpenContributorModal] = useState(false);
+  const [unverifiedAuthor, setUnverifiedAuthor] = useState<UnverifiedContributor | null>(null);
+  const orderedAuthors = [...contributors].map((contributor, index) => ({ ...contributor, sequence: index + 1 }));
+
+  const toggleContributorModal = () => {
+    if (unverifiedAuthor) {
+      setUnverifiedAuthor(null);
+    }
+    setOpenContributorModal(!openContributorModal);
+  };
+
+  useEffect(() => {
+    if (unverifiedAuthor) {
+      // Open modal if user has selected an unverified author
+      setOpenContributorModal(true);
+    }
+  }, [unverifiedAuthor]);
+
+  const handleOnRemove = (indexToRemove: number) => {
+    const remainingAuthors = [...contributors]
+      .filter((_, index) => index !== indexToRemove)
+      .map((contributor, index) => ({ ...contributor, sequence: index + 1 }));
+    setValues(
+      deepmerge(values, { entityDescription: { contributors: remainingAuthors } }, { arrayMerge: overwriteArrayMerge })
```

Comment:
vet du hvorfor vi gjorde det slik i stedet for å bare bruke `remove` i `FieldArray`?
eller evt bare setFieldValue for contributors-feltet

Reply:
`remove` i FieldArray var tungvint å bruke. Den bare fjernet et element i lista, og gjorde ikke noe mer. Og siden vi sorterer på `sequence`, så ble det sånn at hvis du fjernet nr 2 i en liste med sequence på `[1,2,3]`, så fikk du `[1,3]`

Reply:
men det `deepmerge`-greia vet jeg ikke om vi trenger nei. Det var vel du som introduserte det?

Reply:
husker jeg tok inn deepmmerge ja, men husker ikke hvorfor... veldig positiv til å fjerne den hvis den ikke trengs :p 
virker rart å gjøre det slik vi har gjort her, ser ingen grunn til å bruke `setValues` i stedet for `setFieldValue(CONTRIBUTORS_FIELD, remainingAuthors )` her

Reply:
kan sjekke om det gjør noen forskjell med eller uten deepmerge 🤔

Reply:
må sikkert ha deepmerge hvis vi bruker `setValues`, men tror vi slipper unna hvis vi bruker `setFieldValue` i stedet

Reply:
yes, sir - that is correct